src/tracks/playabletrack/wavetrack/ui/ChangeClipSpeedDialog.h

```diff
@@ -0,0 +1,38 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/**********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file ChangeClipSpeedDialog.h
+
+ Dmitry Vedenko
+
+ **********************************************************************/
+#pragma once
+
+#include "wxPanelWrapper.h"
+#include "WaveTrack.h"
+
+class ShuttleGui;
+
+class ChangeClipSpeedDialog final : public wxDialogWrapper
+{
+public:
+   ChangeClipSpeedDialog(
+      WaveTrack& track, WaveTrack::Interval& interval,
+      wxWindow* parent = nullptr);
+
+   ~ChangeClipSpeedDialog() override;
+
+private:
+   void PopulateOrExchange(ShuttleGui& s);
+
+   void OnOk();
+
+   bool ApplyStretchRatio();
```

Comment:
`ApplyStretchRatio` has so far meant rendering time-stretching. It took me some time to realize, though, that this is not this method's intent. Please consider something else, something like `DoChangeClipSpeed` ?